From cfdd64546a87eca30a059f1cfff1e9d8083401d2 Mon Sep 17 00:00:00 2001
From: Sioryn Indianna Nereyeus  Willett <s4392620@student.uq.edu.au>
Date: Mon, 10 Jan 2022 18:08:04 +1000
Subject: [PATCH] cmake now acknowledges vterm.h as the PUBLIC_HEADER

---
 CMakeLists.txt | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4d3a812..195d0d6 100644
--- a/CMakeLists.txt
+++ CMakeLists.txt
@@ -119,6 +119,7 @@ add_library(vterm-static STATIC
     vterm_write.c
     stringv.c
     )
+set_target_properties(vterm-static PROPERTIES PUBLIC_HEADER "vterm.h")
 
 add_library(vterm.o OBJECT
     color_cache.c
@@ -182,6 +183,7 @@ add_library(vterm.o OBJECT
     vterm_write.c
     stringv.c
     )
+set_target_properties(vterm-static PROPERTIES PUBLIC_HEADER "vterm.h")
 
 add_library(vterm-shared SHARED
     $<TARGET_OBJECTS:vterm.o>
@@ -329,6 +331,7 @@ install (TARGETS vterm-static vterm-shared
         ARCHIVE DESTINATION lib
         RUNTIME DESTINATION bin
         INCLUDES DESTINATION include
+        PUBLIC_HEADER DESTINATION include
         )
 
 include(CMakePackageConfigHelpers)
@@ -338,8 +341,6 @@ write_basic_package_version_file(
         COMPATIBILITY AnyNewerVersion
 )
 
-install (FILES vterm.h DESTINATION include)
-
 install(EXPORT libvtermTargets
         FILE libvtermTargets.cmake
         NAMESPACE libvterm::
